[Hanif] iP

hanif-kamal opened the pull request:
# Duke

> "Use Duke, is very good!! - Albert Einstein, mayhaps" 

Do you procrastinate all the time? Is your life an absolute trainwreck?

Get back on track with Duke!

It helps you function like a normal human being by helping you remember to do things!

Duke is:

* text-based
* easy to learn
* ~~FAST~~ <i>SUPER</i> FAST to use

All you need to do is: 

1. download it from [here](https://github.com/hanif-kamal/ip).
1. double-click it.
1. add your tasks
1. let it manage your tasks for you 😉

And it is <b>FREE</b>!

## Features: 
- [x] Add tasks
- [x] Delete tasks
- [ ] Reminders (coming soon!)

If you are also a nerd, here be some code:

```java
public static void main(String[] args) {
        Duke duke = new Duke("./data/duke.txt");
        duke.begin();
}
```

Commit:
Add Gradle support

hanif-kamal committed:
Implement skeletal version of Duke

hanif-kamal committed:
Implement ability to add and list tasks

hanif-kamal committed:
Add Task class

hanif-kamal committed:
Add ability to mark tasks as done

hanif-kamal committed:
Add Deadline, Event and Todo classes

hanif-kamal committed:
Rename variable in Event class

hanif-kamal committed:
Add ToDo, Deadline and Event functionality

hanif-kamal committed:
Add test cases and expected output

hanif-kamal committed:
Add DukeException class

hanif-kamal committed:
Add error handling for incorrect inputs

hanif-kamal committed:
Add delete task functionality and rearrange code structure

hanif-kamal committed:
Add test cases for delete task

hanif-kamal committed:
Change constructors for Deadline, Event, Task and Todo classes

hanif-kamal committed:
Add save and load functionality for tasks

hanif-kamal committed:
Change constructors for Deadline, Event, Task and Todo classes

hanif-kamal committed:
Implement dates and times for Deadline and Event commands

hanif-kamal committed:
Merge branch 'branch-level-7'

hanif-kamal committed:
Merge branch 'branch-level-8'

hanif-kamal committed:
Edit .gitignore

hanif-kamal committed:
Add Storage class

hanif-kamal committed:
Refactor code to utilise Storage class

hanif-kamal committed:
Create TaskList class

hanif-kamal committed:
Refactor code to utilise TaskList class

hanif-kamal committed:
Create Ui class

hanif-kamal committed:
Create Parser class

hanif-kamal committed:
Refactor code to utilise Parser class

hanif-kamal committed:
Divide classes into packages

hanif-kamal committed:
Change loop condition

hanif-kamal committed:
Add JUnit tests for Todo class

hanif-kamal committed:
Add JUnit tests for Deadline and Event classes

hanif-kamal committed:
Swap expected and actual behaviour in test cases

hanif-kamal committed:
Adjust spacing for parsing Deadlines and Events

hanif-kamal committed:
Change import statements

hanif-kamal committed:
Add manifest

hanif-kamal committed:
Modify file paths in runtest.bat

hanif-kamal committed:
Restructure Parser class to use switch statements and string splitting

hanif-kamal committed:
Implement ability to find tasks in a list

hanif-kamal committed:
Add JavaDoc header comments

hanif-kamal committed:
Tweak code to comply with given coding standard

hanif-kamal committed:
Merge branch 'branch-A-JavaDoc'

hanif-kamal committed:
Merge branch 'branch-Level-9' and resolve conflicts

hanif-kamal committed:
Merge branch 'branch-A-CodingStandard' and resolve conflicts

hanif-kamal committed:
Tweak code further to comply with given coding standard

hanif-kamal committed:
Merge branch 'add-gradle-support' of https://github.com/hanif-kamal/ip into branch-A-Gradle

hanif-kamal committed:
Merge branch 'branch-A-Gradle'

hanif-kamal committed:
Add checkstyle

hanif-kamal committed:
Resolve checkStyle errors

hanif-kamal committed:
Add JavaFX dependencies

Zenlzb commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,41 @@
+    /**
+     * Prints a welcome message.
+     */
+    public void welcome() {
```
Should this be printWelcomeMessage?

hanif-kamal replied:
Yes, it should! Thank you very much for spotting this error. I have endeavoured to fix my mistake.

Zenlzb commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,41 @@
+    /**
+     * Prints a goodbye message.
+     */
+    public void bye() {
```
Should this be printGoodbyeMessage?

hanif-kamal replied:
Yes, it should! Thank you very much for spotting this error. I have endeavoured to fix my mistake.

Zenlzb commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,163 @@
+     */
+    public void parse(String input) throws DukeException {
+        String[] splitInput = input.split(" ", 2);
+        String first = splitInput[0];
```
Should this be firstWord or firstParameter?

hanif-kamal replied:
Yes, it should! Thank you very much for spotting this error. I have endeavoured to fix my mistake.

Zenlzb commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,107 @@
+            BufferedReader br = new BufferedReader(new FileReader("./data/duke.txt"));
+            String line;
+            while ((line = br.readLine()) != null) {
+                String[] splitLine = line.split(" ");
```
Should this be lineParts? splitLine sounds like a method

hanif-kamal replied:
Yes, it should! Thank you very much for spotting this error. I have endeavoured to fix my mistake.

Zenlzb reviewed (commented):
cool

hanif-kamal committed:
Begin working on GUI

hanif-kamal committed:
Change image file names

hanif-kamal committed:
Implement GUI and resolve CheckStyle errors

hanif-kamal committed:
Merge branch 'branch-Level-10'

hanif-kamal committed:
Improve formatting to adhere to coding standard

hanif-kamal committed:
Add assertions to document important assumptions

Assertions are not used.

Assertions help to document important assumption.
They are very useful for debugging and testing.

Let's add assertions at some points in the code to better track the program logic.

hanif-kamal committed:
Refactor code to improve code quality

There are some unnecessary lines of code, which reduces the overall code quality.

Having low quality code decreases readability, and may make debugging and testing much more of a hassle.

Let's increase code quality by eliminating unnecessary code.

hanif-kamal committed:
Merge pull request #1 from hanif-kamal/branch-A-Assertions

Add assertions to document important assumptions

hanif-kamal committed:
Merge branch 'master' of https://github.com/hanif-kamal/ip

* 'master' of https://github.com/hanif-kamal/ip:
  Add assertions to document important assumptions

hanif-kamal committed:
Add assertions to document important assumptions

Assertions are not used.

Assertions help to document important assumption.
They are very useful for debugging and testing.

Let's add assertions at some points in the code to better track the program logic.

hanif-kamal committed:
Merge pull request #2 from hanif-kamal/branch-A-CodeQuality

Refactor code to improve code quality

hanif-kamal committed:
Merge branch 'master' of https://github.com/hanif-kamal/ip

* 'master' of https://github.com/hanif-kamal/ip:
  Refactor code to improve code quality

hanif-kamal committed:
Add Notes extension

hanif-kamal committed:
Merge branch 'branch-D-Notes'

hanif-kamal committed:
Fix CheckStyle error

hanif-kamal committed:
Improve GUI

hanif-kamal committed:
Fix bug

hanif-kamal committed:
Set theme jekyll-theme-midnight

hanif-kamal committed:
Add User Guide

hanif-kamal committed:
Merge branch 'master' of https://github.com/hanif-kamal/ip

hanif-kamal committed:
Add Duke title

hanif-kamal committed:
Add Duke screenshot

hanif-kamal committed:
Delete Duke Screenshot

hanif-kamal committed:
Add Duke Screenshot with proper file name

hanif-kamal committed:
Refactor Parser to reduce method length

Fix incorrect exception message for event

hanif-kamal committed:
Add suppressions.xml

Fix checkStyle errors

hanif-kamal committed:
Merge pull request #5 from hanif-kamal/resubmission-updates

Refactor Parser to reduce method length

hanif-kamal committed:
Tweak Exception message slightly to increase readability
